src/NUnitFramework/framework/Internal/Execution/WorkItem.cs

```diff
@@ -480,7 +481,18 @@ private void RunOnSeparateThread()
                 RunOnCurrentThread();
             });
 #if APARTMENT_STATE
-            thread.SetApartmentState(apartment);
+            try
+            {
+                thread.SetApartmentState(apartment);
+            }
+            catch (PlatformNotSupportedException)
+            {
+                string msg = "Apartment state cannot be set on this platform.";
+                log.Error(msg);
+                Result.SetResult(ResultState.Skipped, msg);
+                WorkItemComplete();
+                return;
+            }
```

Comment:
Note the comments around line 220 of this class:
```
            // 2. If the test needs to run in a different apartment.
            // This should not normally occur when using the parallel
            // dispatcher because tests are dispatches to a queue that
            // matches the requested apartment. Under the SimpleDispatcher
            // (--workers=0 option) it occurs routinely whenever a
            // different apartment is requested.
```

This looks like a good fix for sequential tests, but I don't think it covers parallel tests.

Reply:
This was catching the parallel case you mentioned in your other comment.

Reply:
Ah, I understand - `needsNewThread` is being used to create a new thread for tests which would normally be run directly on the Worker. Thanks for explaining - I agree it could do with a comment, I think it really adds a third case to the above! 🙂